### PR TITLE
DROOLS-6540 DMN string join FEEL function

### PR DESCRIPTION
**Pull request by tarilabs**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6540

**referenced Pull Requests**: none

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by tarilabs** [comment]
```
13:25:00  [INFO] --- maven-javadoc-plugin:3.0.1:jar (default) @ drools-workbench-models-guided-dtable ---
Resuming build at Wed Aug 04 11:29:00 UTC 2021 after Jenkins restart
Waiting to resume part of KIE » main » pullrequest » drools-main.pr #72: Finished waiting
```
🤦 

Jenkins retest this

**Comment by tarilabs** [comment]
Jenkins run cdb

**Comment by tarilabs** [comment]
Jenkins run cdb

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/KieFEELExtendedFunctionsTest.java [hunk 1]
```diff
@@ -34,6 +34,17 @@
...
+                { "string join([\"a\",\"b\",\"c\"], null)", "abc", null},
+                { "string join([\"a\"], \"X\")", "a", null},
+                { "string join([\"a\",null,\"c\"], \"X\")", "aXc", null},
+                { "string join([], \"X\")", "", null},
+                { "string join([\"a\",\"b\",\"c\"])", "abc", null},
+                { "string join([\"a\",null,\"c\"])", "ac", null},
+                { "string join([])", "", null},
+                { "string join([\"a\",123,\"c\"], null)", null, FEELEvent.Severity.ERROR},
+                { "string join(null, null)", null, FEELEvent.Severity.ERROR},
                 { "nn sum( 10, null, 20, 40, null )", new BigDecimal("70", MathContext.DECIMAL128), null },
```
Comment (pibizza) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/KieFEELExtendedFunctionsTest.java:48]: I like the declarative style of this test. My concern is that we are mixing different types of expressions - and so different classes to verify - in the same test.

I would suggest to consider splitting this test into separate test classes, for the various features we are considering. I expect these cases are going to grow a lot, and the class would become large.

If the splitting is not possible for technical reasons, I am happy with the patch.

Comment (tarilabs) [reply kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/KieFEELExtendedFunctionsTest.java:48]: Going forward the idea is to actually not have these in a JUnit test at all, and have the "source of the declarative test" in a YAML file or similar, in order to generate the JUnit based on pure data to be used for the current scope, and extend it to covering GWT-translated engine results.

But that is something which requires significative efforts.

As of today, I am continuing to leverage the currently existing suite of test "framework".

**Review by pibizza — APPROVED** [review]

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/KieFEELExtendedFunctionsTest.java (hunk 1 above)
Comment (tarilabs) [reply kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/KieFEELExtendedFunctionsTest.java:48]: /cc @danielezonca

**Review by danielezonca — APPROVED** [review]

**Pull request merged**